solutions/cf-920E.mdx

```diff
@@ -0,0 +1,233 @@
+**Time Complexity:** $\mathcal{O}(N\log N+M)$
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+unordered_set<int> adj[200000];
+//the set of nodes that have not been visited
+set<int> unvis;
+// sz[i] = size of the ith connected component
+int sz[200000];
+//current amount of groups
+int cur = 0;
+void dfs(int x){
+	sz[cur]++;
+	auto it = unvis.begin();
+	while(it != unvis.end()){
+		//if there is no edge between x and *it, skip
+		if(adj[x].count(*it)){
+			it++;
+		}else{
+			//there is a edge between last and x
+			int last = *it;
+			//note it = unvis.erase(it) doenst work here because it could be erased later
+			unvis.erase(it);
+			dfs(last);
+			//find the node after last
+			it = unvis.upper_bound(last);
+		}
+	}
+}
+int main(){
+	int n, m;
+	cin >> n >> m;
+	for(int i = 0; i < m; i++){
+		int a, b;
+		cin >> a >> b;
+		a--;
+		b--;
+		adj[a].insert(b);
+		adj[b].insert(a);
+	}
+	for(int i = 0; i < n; i++){
+		unvis.insert(i);
+	}
+	for(int i = 0; i < n; i++){
+		auto it = unvis.find(i);
+		if(it != unvis.end()){
+			unvis.erase(it);
+			dfs(i);
+			cur++;
+		}
+	}
+	cout << cur << endl;
+	sort(sz, sz + cur);
+	for(int i = 0; i < cur; i++){
+		cout << sz[i] << ' ';
+	}
+	cout << endl;
+}
+```
+</CPPSection>
+</LanguageSection>
+
+## Solution 2 - Selective DFS
+
+Let $N =$ the number of nodes in the graph and $n =$ an abitrary node 
+
+$\texttt{degree}(n)=$ the number of neighbooring nodes (number of different nodes $n$ is directly connected to). 
+
+
+We know that the size of the component containing node $n$ must be at least
+$\texttt{degree}(n)+1$ ($\texttt{degree}(n)$ doesn't include $n$).
+
+If $\texttt{degree}(n_1)+1>\dfrac{N}2$ and $\texttt{degree}(n_2)+1>\dfrac{N}2$, 
+$n_1$ and $n_2$ must intersect, otherwise the graph must be greater than $N$.  
+
+Given this, we can just put all nodes with $\texttt{degree} +1 > \dfrac{N}2$
+into the same group in $O(N)$ time.
+
+Now we can just process all nodes with $\texttt{degree}+1\le\dfrac{N}2$, with DFS.
+
+### Proof
+
+We can handle all nodes with $\texttt{degree}+1>\dfrac{N}2$ in $O(N)$ time. 
+
+Given that $N'$ nodes have at most $\dfrac{N}2 -1 ≈ \dfrac{N}2$ edges, they also must have at least $\dfrac{N}2$ non-edges. Therefore, there must be at least $\dfrac{N'\cdot\dfrac{N}2}2 = \dfrac{N\cdot N'}4$ non-edges (each edge accounts for two nodes).
+
+The input states that the number of edges is less than $2 \cdot 10^5$, therefore $\dfrac{N \cdot N'}4\le2\cdot10^5$, therefore $N \cdot N' \le8\cdot10^5$. 
+
+Doing a simple DFS and checking whether two nodes intersect would be $\mathcal{O}(NN'+ \log M)$, which would fit under the time limit. The $\log M$ comes from processing input. Therefore, this runs under the time limit.
+
+
+<LanguageSection>
+
+<CPPSection>
+
+## Implementation
+**Time Complexity:** $\mathcal{O}(NN'+ \log M)$
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+using vi = vector<int>;
+#define pb push_back
+
+unordered_set<int> adj[200005];
```

Comment:
1. you used 200k for the previous sol, why not 200k here as well?
2. also same thing, use a constant please